src/webgpu/api/operation/buffers/map.spec.ts

```diff
@@ -99,6 +98,73 @@ g.test('mapAsync,write')
     t.checkMapWrite(buffer, rangeOffset, arrayBuffer, rangeSize);
   });
 
+g.test('mapAsync,write,unchanged_ranges_preserved')
+  .desc(
+    `Use mappedAtCreation or mapAsync to write to various ranges of variously-sized buffers, then
+use mapAsync to map a different range and zero it out. Finally use expectGPUBufferValuesEqual
+(which does copyBufferToBuffer + map-read) to verify that contents originally written outside the
+second mapped range were not altered.`
+  )
+  .params(u =>
+    u
+      .beginSubcases()
+      .combine('mappedAtCreation', [false, true])
+      .combineWithParams([
+        { size: 12, range1: [], range2: [8], verify: [0, 8] },
+        { size: 12, range1: [], range2: [0, 8], verify: [8] },
+        { size: 12, range1: [0, 8], range2: [8], verify: [0, 8] },
+        { size: 12, range1: [8], range2: [0, 8], verify: [8] },
+        { size: 28, range1: [], range2: [8, 8], verify: [0, 8] },
+        { size: 28, range1: [], range2: [8, 8], verify: [16] },
```

Comment:
nit: The three regions make this a little hard to reason about (to make sure the right ranges are verified in each test case), and also mean you have to have two cases like this to verify before and after the second range. Maybe you could emulate all of the updates in the "expected" buffer and just check the entire buffer at the end of the test? up to you though, coverage LGTM as is.

Reply:
Refactored, thanks!